server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -74,11 +73,7 @@ public class DesiredBalanceComputer {
     private TimeValue progressLogInterval;
     private long maxBalanceComputationTimeDuringIndexCreationMillis;
 
-    public DesiredBalanceComputer(ClusterSettings clusterSettings, ThreadPool threadPool, ShardsAllocator delegateAllocator) {
-        this(clusterSettings, delegateAllocator, threadPool::relativeTimeInMillis);
-    }
-
-    DesiredBalanceComputer(ClusterSettings clusterSettings, ShardsAllocator delegateAllocator, LongSupplier timeSupplierMillis) {
+    public DesiredBalanceComputer(ClusterSettings clusterSettings, LongSupplier timeSupplierMillis, ShardsAllocator delegateAllocator) {
```

Comment:
Nit: lets keep the original order of arguments (clusterSettings, delegateAllocator, timeSupplierMillis), this could allow to minimize some of the changes such as:
```
            new DesiredBalanceComputer(clusterSettings, shardsAllocator, time::get) {
            new DesiredBalanceComputer(clusterSettings, time::get, shardsAllocator) {
```

Reply:
This IS the original order (i.e. the existing public constructor), just replaced the threadpool with time supplier. :)